### PR TITLE
Updating bigint with FromStr trait for issue 81

### PR DESCRIPTION
**Pull request by josediegorobles**
This is my intent for issue 81
https://github.com/LNP-BP/rust-amplify/issues/81

### CODE REVIEW

File: num/src/bigint.rs [hunk 1]
```diff
@@ -16,6 +16,7 @@
 // If not, see <https://opensource.org/licenses/MIT>.
 
 use crate::error::ParseLengthError;
+use std::convert::TryInto;
```
Comment (dr-orlovsky) [inline num/src/bigint.rs:19]: we need to use `core::` here to maintain `no_std` compatibility

File: num/src/bigint.rs [hunk 2]
```diff
@@ -830,6 +831,18 @@ macro_rules! construct_bigint {
             }
         }
 
+        impl ::core::str::FromStr for $name {
+            #[inline]
+            fn from_str(s: &str) -> Result<Self, Self::Err> {
```
Comment (dr-orlovsky) [inline num/src/bigint.rs:836]: Not sure I understand exactly what the code below does, so test case will be appreciated.

My understanding is that for FromStr implementation we support only decimals (like standard library) and should have `from_str_radix` for other bases (which may be another PR, not necessary doing it here). Second, what you need is not a map over split iterator but a map over `chunks` iterator...

**Review by dr-orlovsky — CHANGES_REQUESTED** [review]
Pls see my comments in the code

File: num/src/bigint.rs (hunk 1 above)
Comment (josediegorobles) [reply num/src/bigint.rs:19]: Ok, but I see std in another partes of the code, getters for example

File: num/src/bigint.rs (hunk 2 above)
Comment (josediegorobles) [reply num/src/bigint.rs:836]: ok, i'm seeing

File: num/src/bigint.rs (hunk 1 above)
Comment (josediegorobles) [reply num/src/bigint.rs:19]: I'm trying to figure out reading u64 Code but I'm not sure how to proceed.
If you can give me an indication I would be pleased

Comment (dr-orlovsky) [reply num/src/bigint.rs:19]: Getters are part of other crate, which is not `no_std` one. The main indicator is when CI is broken, and here it indeed is.

Comment (dr-orlovsky) [reply num/src/bigint.rs:19]: > I'm trying to figure out reading u64 Code but I'm not sure how to proceed.

not sure I understood your question

**Comment by dr-orlovsky** [comment]
We need to fix CI and add tests here. Please let me know if you would like to proceed with this PR further

**Comment by josediegorobles** [comment]
> We need to fix CI and add tests here. Please let me know if you would like to proceed with this PR further

Ok, i will see tonight or tomorrow. I will continue if you think it's worth it
